src/xpk/commands/job.py

```diff
@@ -14,12 +14,87 @@
 limitations under the License.
 """
 
+from ..core.commands import run_command_for_value, run_command_with_updates
 from ..utils.console import xpk_exit, xpk_print
-from ..core.core import add_zone_and_project
 from ..core.app_profile import APP_PROFILE_TEMPLATE_DEFAULT_NAME
-from ..core.commands import (
-    run_command_with_updates,
-)
+from ..core.core import add_zone_and_project
+from ruamel.yaml import YAML
+import re
+import sys
+
+
+def job_info(args):
+  """Run commands obtaining information about a job given by name.
+
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    0 if successful and 1 otherwise.
+  """
+  job_name = args.name
+
+  desc_command = f'kubectl-kjob describe slurm {job_name}'
+  desc_code, desc_text = run_command_for_value(
+      desc_command, 'Getting job data', args
+  )
+  if desc_code != 0:
+    xpk_print(f'Data info request returned ERROR {desc_code}')
+    xpk_exit(desc_code)
+
+  job_command = (
+      'kubectl-kjob list slurm -o yaml --field-selector'
+      f' metadata.name=={job_name}'
+  )
+  job_code, job_text = run_command_for_value(
+      job_command, 'Getting job info', args
+  )
+  if job_code != 0:
+    xpk_print(f'Job info request returned ERROR {job_code}')
+    xpk_exit(job_code)
+
+  pods_command = f'kubectl get pods -l=job-name={job_name} --no-headers'
+  pods_code, pods_text = run_command_for_value(
+      pods_command, 'Getting pods list', args
+  )
+  if pods_code != 0:
+    xpk_print(f'Pods list request returned ERROR {pods_code}')
+    xpk_exit(pods_code)
+
+  yaml = YAML(typ='safe')
+  job_yaml = yaml.load(job_text)['items'][0]
+
+  output = {
+      'Job name': job_name,
+      'Profile': get_profile(job_yaml),
+      'Labels': job_yaml['metadata']['labels'],
+      'Mounts': job_yaml['spec']['template']['spec']['containers'][0][
+          'volumeMounts'
+      ],
+      'Environment variables': get_ev_vars(desc_text),
+      'Pods': get_pods(pods_text),
+  }
+
+  yaml.default_flow_style = False
+  yaml.sort_keys = False
+  yaml.dump(output, sys.stdout)
+
+
+def get_profile(job_yaml: dict) -> str:
+  env_vars = job_yaml['spec']['template']['spec']['containers'][0]['env']
+  profile = next((x['value'] for x in env_vars if x['name'] == 'PROFILE'), '')
+  return profile
+
+
+def get_ev_vars(job_desc_text: str) -> list[tuple[str, str]]:
+  regex = r'(SLURM_[A-Z_]*=.*)'
+  search_res = re.findall(regex, job_desc_text)
+  return search_res
```

Comment:
It will not work well. We can't see env variables with describe, because it is generating on runtime and it is different for each pod.

Reply:
Here is and example of the output of this command (only the relevant part):
```
- SLURM_ARRAY_JOB_ID=1
- SLURM_ARRAY_TASK_COUNT=3
- SLURM_ARRAY_TASK_MAX=3
- SLURM_ARRAY_TASK_MIN=1
- SLURM_TASKS_PER_NODE=1
- SLURM_CPUS_PER_TASK=
- SLURM_CPUS_ON_NODE=
- SLURM_JOB_CPUS_PER_NODE=
- SLURM_CPUS_PER_GPU=
- SLURM_MEM_PER_CPU=
- SLURM_MEM_PER_GPU=
- SLURM_MEM_PER_NODE=
- SLURM_GPUS=
- SLURM_NTASKS=1
- SLURM_NTASKS_PER_NODE=1
- SLURM_NPROCS=1
- SLURM_NNODES=2
- SLURM_SUBMIT_DIR=/slurm/scripts
- SLURM_SUBMIT_HOST=$HOSTNAME
- SLURM_JOB_NODELIST=xpk-def-app-profile-slurm-hvz49-0.xpk-def-app-profile-slurm-hvz49,xpk-def-app-profile-slurm-hvz49-1.xpk-def-app-profile-slurm-hvz49
- SLURM_JOB_FIRST_NODE=xpk-def-app-profile-slurm-hvz49-0.xpk-def-app-profile-slurm-hvz49
- SLURM_JOB_ID=$(expr $JOB_COMPLETION_INDEX \* 1 + $i + 1)
- SLURM_JOBID=$(expr $JOB_COMPLETION_INDEX \* 1 + $i + 1)
- SLURM_ARRAY_TASK_ID=$container_index
- SLURM_JOB_FIRST_NODE_IP=${SLURM_JOB_FIRST_NODE_IP:-""}
```

sure some of them are empty / unevaluated expressions, but some are set. There is value in printing it out. 

For running containers I could exec into the container and print out the values of the variables. It would be easy to do for ordinary env variables, but a hassle for this ones since they are written to some file in the filesystem of the pod, not exported into the env itself.

Should I just print out ordinary env variables and only for running pods?
Should I print out these statements, or maybe only the ones which values are known at runtime?
Are they really env variables if they are not being set? Are they even relevant for the user if they are not set in the pod?

Personally I would suggest printing env variables only for running pods, and only the real env variables, set and visible when using printenv. What do you think? Is there value in that?

Reply:
They are not shown using printenv, they sit in files in slurm/env/i (where i in what I used is 0, but I'm not sure if  can count on it). All I'm getting using printenv are:
```
KUBERNETES_SERVICE_PORT_HTTPS, KUBERNETES_SERVICE_PORT, USER_ID, HOSTNAME, PROFILEPWD, JOB_COMPLETION_INDEX, HOME, KUBERNETES_PORT_443_TCP, LS_COLORS, TIMESTAMP, TERM, TASK_NAME, SHLVL, KUBERNETES_PORT_443_TCP_PROTO, JOB_CONTAINER_INDEX, KUBERNETES_PORT_443_TCP_ADDR, KUBERNETES_SERVICE_HOST, KUBERNETES_PORT, KUBERNETES_PORT_443_TCP_PORT, PATH, TASK_ID, OLDPWD, _
```

Reply:
I mean execute printenv on the script file.

Reply:
So I guess I will provide status of the pod, and if it's running I will also print environment variables that are set within the pod (obtainable with printenv)? It seems to me like this is what is usable by user. Or maybe the info from the slurm/env files would also be beneficial?

Reply:
Yes, but for each container it can be different.

Reply:
I'm not sure that we need to show this variables on the describe command. Is it requirementes?

Reply:
> I mean execute printenv on the script file and get it by logs.

Ah we can't log on script.

Reply:
> Or maybe the info from the slurm/env files would also be beneficial?

On that case I think this is only one way. But again, it can be different for each container.